Python/optimizer_analysis.c

```diff
@@ -447,32 +435,22 @@ optimize_uops(
         ctx->frame->stack_pointer = stack_pointer;
         assert(STACK_LEVEL() >= 0);
     }
-    Py_UNREACHABLE();
-
-out_of_space:
-    DPRINTF(3, "\n");
-    DPRINTF(1, "Out of space in abstract interpreter\n");
-    goto done;
-error:
-    DPRINTF(3, "\n");
-    DPRINTF(1, "Encountered error in abstract interpreter\n");
-    if (opcode <= MAX_UOP_ID) {
-        OPT_ERROR_IN_OPCODE(opcode);
+    if (ctx->out_of_space) {
```

Comment:
I wonder if this makes those cases perform worse, since we don't bail early anymore. Then again, this should be quite rare (almost never if i recall from pyperformance).

Reply:
The loop condition is `ctx->done` so we leave the loop immediately we run out of space.